server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -322,6 +324,50 @@ public Iterator<Setting<?>> settings() {
         Property.Final
     );
 
+    /**
+     * Used to specify remote store repository to use for this index.
+     */
+    public static final Setting<String> INDEX_REMOTE_STORE_REPOSITORY_SETTING = Setting.simpleString(
+        SETTING_REMOTE_STORE_REPOSITORY,
+        new Setting.Validator<>() {
+
+            @Override
+            public void validate(final String value) {}
+
+            @Override
+            public void validate(final String value, final Map<Setting<?>, Object> settings) {
+                if (value == null || value.isEmpty()) {
+                    throw new IllegalArgumentException(
+                        "Setting " + INDEX_REMOTE_STORE_REPOSITORY_SETTING.getKey() + " should be provided with non-empty repository ID"
+                    );
+                } else {
+                    validateRemoteStoreSettingEnabled(settings, INDEX_REMOTE_STORE_REPOSITORY_SETTING);
+                }
+            }
+
```

Comment:
Do we need to validate if the repository is valid and exists essentially checking to all the list of registered repositories

Reply:
Yes, we can add that check. Index creation will anyway fail if repository does not exist but it will take time to fail and will not provide helpful error message.

Let me add a check for repository exists.

Reply:
@Bukhtawar To check against actual repository, we need to inject `RepositoriesService` into `IndexMetadata` whose builder/constructor mostly takes index settings related parameters (like `numberOfShards`, `requireFilters`, `indexCreatedVersion`). IMO, RepositoriesService injection would be an outlier. Need to think more on that. What do you think?

Reply:
We might have to support overloaded constructor or builders since I believe now we are associating a new attribute to IndexMetadata?

Reply:
I have created a tracking issue here to handle this: https://github.com/opensearch-project/OpenSearch/issues/4403

Reply:
Got it. Created tracking issue here: